de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/dvlm/mat/MatImporterTest.java

```diff
@@ -0,0 +1,464 @@
+		TestCategoryAllProperty tc = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc);
+		//Create Mat
+		MatFile matfile = Mat5.newMatFile();
+		
+		Struct boolPropertyStruct = Mat5.newStruct();
+		boolPropertyStruct.set(MatHelper.VALUE, Mat5.newLogicalScalar(TEST_BOOL));
+		
+		Struct stringPropertyStruct = Mat5.newStruct();
+		stringPropertyStruct.set(MatHelper.VALUE, Mat5.newString(TEST_STRING));
+		
+		Struct intPropertyStruct = Mat5.newStruct();
+		intPropertyStruct.set(MatHelper.UNIT, Mat5.newString(""));
+		intPropertyStruct.set(MatHelper.VALUE, Mat5.newScalar(TEST_INT));
+		
+		Struct floatPropertyStruct = Mat5.newStruct();
+		floatPropertyStruct.set(MatHelper.UNIT, Mat5.newString(""));
+		floatPropertyStruct.set(MatHelper.VALUE, Mat5.newScalar(TEST_FLOAT));
+		
+		Struct enumPropertyStruct = Mat5.newStruct();
+		enumPropertyStruct.set(MatHelper.UNIT, Mat5.newString("Kilogram"));
+		enumPropertyStruct.set(MatHelper.VALUE, Mat5.newString("25"));
+		enumPropertyStruct.set(MatHelper.NAME, Mat5.newString(TEST_ENUM_VALUE));
+
+		Struct resourcePropertyStruct = Mat5.newStruct();
+		resourcePropertyStruct.set(MatHelper.URI, Mat5.newString(""));
+		
+		Struct caStruct = Mat5.newStruct();
+		caStruct.set("testString", stringPropertyStruct);
+		caStruct.set("testInt", intPropertyStruct);
+		caStruct.set("testFloat", floatPropertyStruct);
+		caStruct.set("testResource", resourcePropertyStruct);
+		caStruct.set("testEnum", enumPropertyStruct);
+		caStruct.set("testBool", boolPropertyStruct);
+		
+		Struct struct = Mat5.newStruct();
+		struct.set(MatHelper.TYPE, Mat5.newString(sei.getType().getName()))
+			  .set(MatHelper.UUID, Mat5.newString(sei.getUuid().toString()));
+		struct.set(tc.getName(), caStruct);
+		matfile.addArray(sei.getName(), struct);
+		Command cmd = importer.importSei(editingDomain, sei, matfile);
+		editingDomain.getCommandStack().execute(cmd);
+		assertEquals("Is the same", TEST_STRING, tc.getTestString());
+		assertEquals("Is the same", TEST_ENUM_VALUE, tc.getTestEnum());
+		assertEquals("Is the same", TEST_BOOL, tc.getTestBool());
+		assertEquals("Is the same", TEST_INT, tc.getTestInt());
+		assertEquals(TEST_FLOAT, tc.getTestFloat(), 0);
+		assertEquals("Is the same", null, tc.getTestResource());
+	}
+	
+	@Test
+	public void testCheckIfCorrectSeiCorrect() {
+		assertTrue("Is the same", importer.checkIfCorrectSei(sei, mat));
+		sei.setName("Testsein");
+		assertFalse("Not the same name", importer.checkIfCorrectSei(sei, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongUUID() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName(TEST_SEI);
+		sei2.setType(sei.getType());
+		assertFalse("Not the same uuid", importer.checkIfCorrectSei(sei2, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongType() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);

```

Comment:
That test is legacy or? It doesn't do anything else than the upper test? Or am I missing something?

Reply:
One checks for UUID, one for TYPE